src/nvim/os/fs.c

```diff
@@ -391,6 +391,37 @@ int os_open(const char* path, int flags, int mode)
   return r;
 }
 
+/// Set a file to close on exec.
+//
+// @return -1 if failed to set, 0 otherwise.
+int os_set_cloexec(const int fd)
+{
+#ifdef HAVE_FD_CLOEXEC
+  int e;
+  int fdflags = fcntl(fd, F_GETFD);
+  if (fdflags < 0) {
+    e = errno;
+    ELOG("Failed to get flags on descriptor %d: %s", fd, strerror(errno));
```

Comment:
Can I just double check I understand -- (I think I do, but asking for clarification seems like it may save some time if I'm wrong).

Were you originally saying that hypothetically, if the `ELOG` macro could set `errno`, then we would have to use `e` in the call to `strerror()`. Then in the second comment saying that since it doesn't, we're fine as we are.

Or were you working on the thought that I was using the temporary variable to ensure the errno passed to `strerror()` was correct, and pointing out that that wasn't ensured by what I've done.

I think you meant the former, but that may be because I'm coloured by knowing what I intended.

What I intended to ensure here, was that whether or not `strerror()` changes `errno`, the `errno` set when `os_set_cloexec()` returns is the `errno` that relates to the `fcntl()` call (in case any user wants to use it).

While checking just now if it's possible that `strerror()` changes `errno`, I noticed that in [posix 2004](http://pubs.opengroup.org/onlinepubs/009695399/,) the return value of `strerror()` when an error occurs isn't specified, so it could be `NULL`. I'll add a check for that.